### PR TITLE
feat: paypal multiparty changes

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: components/checkout/PaymentStep/PaymentStep.tsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ const PaymentStep = (props: PaymentStepProps) => {
   const { loadPaymentTypes } = usePaymentTypes()
   const paymentTypes = loadPaymentTypes()
   const { validateCustomerAddress } = useValidateCustomerAddress()
-  const paypalBearerToken = usePayPalBearerToken()
+  const isPayPalEnabled = useIsPayPalEnabled()
```
Comment (kibo-chandradeeptalaha) [inline components/checkout/PaymentStep/PaymentStep.tsx:163]: I think the hook name should be usePaypal() that should expose the isEnabled property.

File: hooks/custom/useIsPayPalEnabled/useIsPayPalEnabled.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ */
+export const useIsPayPalEnabled = () => {
+  const [isPayPalEnabled, setIsPayPalEnabled] = useState(null)
+
+  useEffect(() => {
+    const checkIsPayPalEnabled = async () => {
+      try {
+        const url = `${
+          process.env.NEXT_PUBLIC_URL ? process.env.NEXT_PUBLIC_URL : ''
```
Comment (kibo-chandradeeptalaha) [inline hooks/custom/useIsPayPalEnabled/useIsPayPalEnabled.tsx:22]: I don't think we need this conditional render, Next should handle that api route calling by itself

**Review by kibo-chandradeeptalaha — CHANGES_REQUESTED** [review]
Please take a look at these minor things. Otherwise looks good

File: hooks/custom/useIsPayPalEnabled/useIsPayPalEnabled.tsx (hunk 2 above)
Comment (kibo-sushant) [reply hooks/custom/useIsPayPalEnabled/useIsPayPalEnabled.tsx:22]: We are following same syntax everywhere, Just added it for consistency purpose.
export function makeGraphQLClient(endpoint?: string) {
  endpoint =
    endpoint || `${process.env.NEXT_PUBLIC_URL ? process.env.NEXT_PUBLIC_URL : ''}/api/graphql`
  return new GraphQLClient(endpoint)
}

File: components/checkout/PaymentStep/PaymentStep.tsx (hunk 1 above)
Comment (kibo-sushant) [reply components/checkout/PaymentStep/PaymentStep.tsx:163]: As there are multiple hooks related to PayPal (usePayPalBearerToken, useIsPayPalEnabled) hence we cannot keep generic name

**Review by Kibo-TriptiV — APPROVED** [review]

**Review by kibo-chandradeeptalaha — APPROVED** [review]

**Pull request merged**